ThreadGroup/Controllers/TGMainViewController.m

```diff
@@ -412,6 +421,7 @@ - (void)TGDeviceStepView:(TGDeviceStepView *)stepView didTapIcon:(id)sender {
     if (stepView == self.wifiSearchView) {
         [[UIApplication sharedApplication] openURL:[NSURL URLWithString:UIApplicationOpenSettingsURLString]];
     } else if (stepView == self.routerSearchView) {
+        self.shouldCancelRouterConnection = YES;
         [self.selectDeviceView resignFirstResponder];
         [self setViewState:TGMainViewStateLookingForRouters];
```

Comment:
You want to be setting `self.shouldCancelRouterConnection = NO` in the `-connectToRouter` method.